### PR TITLE
Config: Remove unused WooCommerce extension flags

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

This PR removes some unused feature flags, mostly related to the Woo extension that we recently have been deprecating and cleaning up in #51012, #52819, and #51309. We're also removing a couple of legacy post editor flags.

#### Testing instructions

Verify that the removed feature flags are not in use.

### CODE REVIEW

**Review by noahtallen — APPROVED** [review]

File: config/development.json [hunk 1]
```diff
@@ -193,24 +192,7 @@
...
 		"upsell/nudge-component": true,
 		"woocommerce/extension-dashboard": true,
-		"woocommerce/extension-dashboard-stats-widget": true,
-		"woocommerce/extension-orders": true,
-		"woocommerce/extension-orders-create": true,
-		"woocommerce/extension-orders-edit": true,
-		"woocommerce/extension-products": true,
-		"woocommerce/extension-product-categories": true,
-		"woocommerce/extension-promotions": true,
 		"woocommerce/extension-referrers": true,
```
Comment (jsnajdr) [inline config/development.json:195]: `referrers` is a project that never got finished and remained enabled only in development. A follow-up PR should either remove it, or enable it for all if it's good enough.

File: config/development.json [hunk 2]
```diff
@@ -143,7 +143,6 @@
 		"perfmon": false,
 		"plans/personal-plan": true,
 		"post-editor/checkout-overlay": true,
```
Comment (jsnajdr) [inline config/development.json:145]: The `checkout-overlay` is enabled everywhere but Desktop. Intent or oversight?

File: config/development.json [hunk 3]
```diff
@@ -193,24 +192,7 @@
 		"upsell/concierge-session": true,
 		"upsell/nudge-component": true,
 		"woocommerce/extension-dashboard": true,
```
Comment (jsnajdr) [inline config/development.json:194]: `extension-dashboard` is also enabled everywhere but Desktop. Intent or oversight?

File: config/development.json [hunk 4]
```diff
@@ -193,24 +192,7 @@
...
-		"woocommerce/extension-settings-email": true,
-		"woocommerce/extension-settings-email-generic": true,
-		"woocommerce/extension-settings-payments": true,
-		"woocommerce/extension-settings-shipping": true,
-		"woocommerce/extension-settings-stripe-connect-flows": true,
-		"woocommerce/extension-settings-tax": true,
-		"woocommerce/extension-wcservices": true,
-		"woocommerce/extension-wcservices/international-labels": true,
 		"woocommerce/onboarding-oauth": true,
 		"woocommerce/store-on-non-atomic-sites": true,
```
Comment (jsnajdr) [inline config/development.json:197]: `store-on-non-atomic-sites` is never read. Can be removed.

File: config/development.json [hunk 5]
```diff
@@ -193,24 +192,7 @@
...
-		"woocommerce/extension-settings": true,
-		"woocommerce/extension-settings-email": true,
-		"woocommerce/extension-settings-email-generic": true,
-		"woocommerce/extension-settings-payments": true,
-		"woocommerce/extension-settings-shipping": true,
-		"woocommerce/extension-settings-stripe-connect-flows": true,
-		"woocommerce/extension-settings-tax": true,
-		"woocommerce/extension-wcservices": true,
-		"woocommerce/extension-wcservices/international-labels": true,
 		"woocommerce/onboarding-oauth": true,
```
Comment (jsnajdr) [inline config/development.json:196]: `onboarding-oauth` is always true except Desktop. It enables some custom styles and copy in Woo signup and login flows. Desktop user has zero chance of getting there.

File: config/desktop-development.json [hunk 6]
```diff
@@ -104,7 +104,6 @@
 		"p2/p2-plus": true,
 		"perfmon": false,
 		"plans/personal-plan": true,
-		"post-editor/html-toolbar": true,
 		"post-editor/revisions": true,
```
Comment (jsnajdr) [inline config/desktop-development.json:107]: `post-editor/revisions` is never read -- can be removed.

**Review by jsnajdr — COMMENTED** [review]
Posting some ideas for follow-ups.

**Comment by tyxla** [comment]
Thanks, Jarda, I'm going to follow up to further clean those up in the near future.

**Pull request merged**